File: experiments/higgs/sort/files/listII/mainC.ts
    var intList : IList;
    export function setup()
	{
		intList=makeIntList();
	}
    export function resetup()
	{
		intList=makeIntList();
	}
	export function main()
	{
		test(intList);
	}

	function test(list : IList)
	{
		quicksort(list);
	}
	
	function makeIntList() : IList
	{
		var list : IList = new List(5);
		var i : number = 0;
		while (i < 100000)
		{
			var num : number = (i * 163841 + 176081) % 122251;
			list.add(num);
			i = i + 1;
		}
		return list;
	}

File: experiments/java/sort/files/MainC.java
public class Main
{
	public static void main(String[] args)
	{
		IList<Long> intlist = MakeIntList();
		Main.Test(intlist);
		intlist = MakeIntList();
		Main.Test(intlist);
		intlist = MakeIntList();
		Main.Test(intlist);
		intlist = MakeIntList();
		Main.Test(intlist);
		IIterator<Long> ilistIter = intlist.GetIterator();
		ilistIter.MoveNext();
		long last = ilistIter.Current();
		while(ilistIter.MoveNext())
		{
			if(ilistIter.Current()<last)
			{
				throw new RuntimeException("sorting failed!");
			}
			last = ilistIter.Current();
		}
	}

	public static void Test(IList<Long> list)
	{
		long start = System.nanoTime();
		Sort.Quicksort(list);
		long end=System.nanoTime();
		System.out.print(((double)(end-start))/1000000000);
		System.out.println(" seconds");
	}
	
	private static long longmod(long a, long b)
	{
		return (a < 0) ? (b-1 - (-a-1) % b) : (a % b);
	}
	
	public static IList<Long> MakeIntList()
	{
		IList<Long> list = new ListImpl<Long>(5L);
		long i =0;
		while(i<100000)
		{
			long opnd1 =  (i * 163841 + 176081);
			long num = longmod(opnd1, 122251);
			list.Add(num);
			i=i+1;
		}
		return list;
	}
}
